.eslintignore

```diff
@@ -0,0 +1,57 @@
+# Vendor files and generated test artifacts
+**/vendor
+test_root/staticfiles
+
+
+# Vendor files living outside the /vendor/ dir
+*.min.js
+*-min.js
+*.nocache.js
+**/bootstrap*.js
+**/jquery*.js
+**/d3*.js
+
+
+# Translations files
+**/static/js/i18n
+
+
+# Gitignored xmodule stuff
+common/static/xmodule
+
+
+# Coffeescript directories (don't lint autogenerated files)
+cms/static/coffee
+lms/static/coffee
+common/static/coffee
+common/lib/capa/capa/tests/test_files/js
+
+
+# Symlinks into common/lib/xmodule/xmodule/js
+cms/static/xmodule_js
+lms/static/xmodule_js
+
+
+# This directory is about half Coffee and half JS, things get messy here so just ignore all existing coffee paths
+common/lib/xmodule/xmodule/js/spec/annotatable/display_spec.js
+common/lib/xmodule/xmodule/js/spec/capa/display_spec.js
+common/lib/xmodule/xmodule/js/spec/html/edit_spec.js
+common/lib/xmodule/xmodule/js/spec/problem/edit_spec.js
+common/lib/xmodule/xmodule/js/spec/problem/edit_spec_hint.js
+common/lib/xmodule/xmodule/js/spec/tabs/edit.js
+
+common/lib/xmodule/xmodule/js/src/annotatable/display.js
+common/lib/xmodule/xmodule/js/src/capa/display.js
+common/lib/xmodule/xmodule/js/src/conditional/display.js
+common/lib/xmodule/xmodule/js/src/discussion/display.js
+common/lib/xmodule/xmodule/js/src/html/display.js
+common/lib/xmodule/xmodule/js/src/html/edit.js
+common/lib/xmodule/xmodule/js/src/javascript_loader.js
+common/lib/xmodule/xmodule/js/src/problem/edit.js
+common/lib/xmodule/xmodule/js/src/raw/edit/json.js
+common/lib/xmodule/xmodule/js/src/raw/edit/metadata-only.js
+common/lib/xmodule/xmodule/js/src/raw/edit/xml.js
+common/lib/xmodule/xmodule/js/src/sequence/display.js
+common/lib/xmodule/xmodule/js/src/sequence/edit.js
+common/lib/xmodule/xmodule/js/src/tabs/tabs-aggregator.js
+common/lib/xmodule/xmodule/js/src/vertical/edit.js
```

Comment:
FYI, I've verified that I have all Coffeescript files ignored from eslint by doing a `git clean -fdX`, running ESLint, running `paver compile_coffeescript`, running ESLint, and checking that the violation numbers from the two runs match.